script/check_jdk_version.go

```diff
@@ -0,0 +1,33 @@
+package main
+
+import (
+	"fmt"
+	"os/exec"
+	"regexp"
+	"strconv"
+)
+
+func main() {
+
+	cmd := exec.Command("java", "-version")
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		panic(fmt.Sprintf("Error: %v\n", err))
+	}
+
+	versionStr := string(output)
+	versionRegex := regexp.MustCompile(`version "?([1-9]+)(\.([0-9]+)){2,3}"?`)
+	matches := versionRegex.FindStringSubmatch(versionStr)
+	if len(matches) < 2 {
+		panic(fmt.Sprintf("Unable to determine Java version: %s\n", versionStr))
```

Comment:
IMO we should not panic. We cannot test all the possible JDK out there to make sure they return the expected text.

Reply:
ok. How about we get the java version from maven i.e.

```
$ mvn -version
Apache Maven 3.9.6 (bc0240f3c744dd6b6ec2920b3cd08dcc295161ae)
Maven home: /opt/homebrew/Cellar/maven/3.9.6/libexec
Java version: 17.0.6, vendor: Azul Systems, Inc., runtime: /Library/Java/JavaVirtualMachines/zulu-17.jdk/Contents/Home
Default locale: en_GB, platform encoding: UTF-8
OS name: "mac os x", version: "14.4.1", arch: "aarch64", family: "mac"
```

Reply:
Okey. Still make sure any parsing problem is not leading to a panic which would impede to complete a build.